R/globals.R

```diff
@@ -25,6 +25,32 @@ register_s3_method <- function(pkg, generic, class, fun = NULL) {
   )
 }
 
+register_upgrade_message <- function(pkg, version) {
+  # Do nothing if the package isn't installed or
+  # meets the version requirement
+  if (system.file(package = pkg) == "") return(NULL)
+  if (packageVersion(pkg) >= as.character(version)) return(NULL)
```

Comment:
Sorry, I don't believe this is the right place to check; don't we want to check right before each of the two `packageStartupMessage(msg)` lines?

Reply:
Much better, thanks. I'm running some spot tests now, if it all looks good I'll approve